conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/RequestContext.java

```diff
@@ -49,4 +51,9 @@
      * Implementations may choose to include this data in the request log.
      */
     void requestArg(Arg<?> arg);
+
+    /**
+     * Returns the client certificates associated with the connection used to make the current request.
+     */
+    Optional<ImmutableList<Certificate>> peerCertificates();
```

Comment:
Why ImmutableList in the API?

Reply:
No strong opinion on this, `ImmutableList` makes it clear that the list cannot be modified and we already use guava types for conjure ListenableFuture as well as `ListMultimap` a few lines above in this file.
List would be fine, mostly wanted to avoid the `Certificate[]` type returned by the `SSLSession`.

Reply:
I shouldn't have used an optional, empty list provides the same information, and consumers will likely already have to check for an empty list.
```suggestion
    ImmutableList<Certificate> peerCertificates();
```

Reply:
I thought about that, but then realized that Optional is not a bad way to signal that not everything will use this feature.

Reply:
Alternatively, maybe you want to actually have a small POJO? Reading the javadoc:

```
Returns:
an ordered array of peer certificates, with the peer's own certificate first followed by any certificate authorities.
```

There's clearly some structure to this. Do we know how people will validate the certs? Maybe some utils would be useful?

Reply:
I think the `RequestContext` itself is a collection of features that most consumers won't need, but unblock specific use cases. In thinking about how this would be used, I'm pretty sure the list empty check will still have to be run even when the optional is present, so at that point list.isEmpty is just as easy to check. I don't know what it would mean if you got a present optional of an empty list, that should be an impossible state, but the api doesn't enforce/restrict it.

Reply:
> Do we know how people will validate the certs? Maybe some utils would be useful?

Since this is a drop-through to the SSLSession, standard tooling should be sufficient and I don't think we enforce opinions.